### ISSUE TITLE
Extension displays `Your host device is not running Tailscale` erroneously [but works just fine]

### ISSUE THREAD

**Issue by QuantumLibet** [tailscale/docker-extension#34]
Let me start with the most obvious: I'm a complete n00b, even more so when it comes to Tailscale. And English is not my mother tongue. The following might be my misinterpretation of something. Please excuse my errors.

## Versions in use
* macOS 10.15.7, SecUpd 2022-004
* Docker Desktop 4.9.0 (80466)
* Tailscale macOS AppStore client 1.24.2
* Tailscale Docker-Extension 0.0.10
* Tailscale Docker-Extention tailscaled v1.22.1, Go 1.17.8, started with '--tun=userspace-networking'

## Other info for completness' sake:
* Services [collection] enabled
* Global Nameserver enabled (as 100.106.200.48); Override local DNS _DIS_abled
* MagicDNS enabled (as always: 100.100.100.100)
* No ACLs set (other than defaults)
* Authentication for all tailscaled's using the same user (vpn@mydomain.tld)

## Observation
After upgrading the Docker Desktop.app from from 4.8.2,79419 to 4.9.0,80466 the Tailscale Docker-Extension claims `Your host device is not running Tailscale` and `No containers are running`.

Both is not really correct.
* The Tailscale AppStore client installed on the host is running just fine, and there are containers running as well.
* The web admin machines console (login.tailscale.com/admin/machines) shows the client, as well as the docker extension as connected.
* The web admin services console (login.tailscale.com/admin/services) shows the ssh service of the client, as well as the various tcp ports of the containers running as available.
* Testing connection to the client works.
* Testing connection to some container works as well. Both with the LAN IP as well as with the tailscale IP and the tailscale FQDN (hostname.mydomain.tld.beta.tailscale.net)

There seems to be no problem (except a "cosmetic" one?)

### Some weird observation:
Docker Desktop.app > Dashboard should show the Tailscale extension in the left sidebar, unter the "Extensions beta" header.

Restarting the DD.app with the [macOS] menu bar Docker icon and choosing "Restart" will drop/hide the Tailscale extension in the Dashboard.
Restarting the DD.app by quitting and opening again, will show the Tailscale extension in the Dashboard again.  O_-

The Tailscale extension container _ is _ running all the time, as can be seen in the DD.app > Dashboard > Containers > tailscale_docker-extension-desktop-extension, it's just the display in the Dashboard that is broken. Which somehow confirms my uneducated assumption that this is a cosmetic problem. It _ does _ say "beta" on the tin.

### Some other peculiar observation:

Quitting and Opening (as opposed to "Restart") the Docker Desktop.app and waiting some, will eventually show the running containers in the Tailscale extension overview.
The error `No containers are running`is then gone. The error `Your host device is not running Tailscale` is still shown.

**Comment by QuantumLibet** [comment]
The update from Tailscale Docker-Extension 0.0.10 to 0.0.11 did not fix the issue.

Could the problem be related to the App Store version of the Tailscale client? The App Store client does not expose the `tailscale` command, which adds it's very own complexity (like following instructions like [this](https://tailscale.com/kb/1019/subnets/?tab=macos#step-2-connect-to-tailscale-as-a-subnet-router).

**Comment by QuantumLibet** [comment]
The update from Tailscale Docker-Extension 0.0.11 to 0.0.12 did not fix the issue.

**Comment by rosszurowski** [comment]
Hi! Sorry for the trouble, and thanks for reporting this. Our "is Tailscale installed" check is a little hacky, since it just checks a particular path. It looks like you're running macOS, [which would run this code](https://github.com/tailscale/docker-extension/blob/main/host/host-tailscale#L5) when checking.

Where is your local copy of Tailscale.app installed? Maybe there are ways we can make our check more robust.

**Comment by QuantumLibet** [comment]
Tailscale has been installed from the App Store and is in the default location of '/Application/Tailscale.app'.

Copying the linked file locally and executing it with `/bin/sh -x ./host-tailscale present` gives no error (as expected). 
Quitting Tailscale and executing the script with `… start` will start Tailscale (as expected). 
Executing `… status` can't be tested here, since I wouldn't know how the script is called and what value `$2` could/would/should have.

Since the script works fine, could it be something in the function calling the script?

**Comment by rosszurowski** [comment]
Hmm, the core block [which runs that script is found here](https://github.com/tailscale/docker-extension/blob/main/ui/src/tailscale.ts#L211-L233).

Reviewing the code again, it looks like it's correctly finding that you have Tailscale installed, it just can't detect that it's currently running. (Hence the "not running Tailscale" language, which could be improved). I think it may be failing [in the status block](https://github.com/tailscale/docker-extension/blob/main/ui/src/tailscale.ts#L222-L231) — either it can't run `tailscale status`, or it's getting an unexpected result back.

Can you run this command, and post the output of a few fields?

```
/Applications/Tailscale.app/Contents/MacOS/tailscale status --json
```

This will return a big JSON blob. Don't post it here, since the endpoint IPs can be private information, but I'm interested in the values of these fields: `Version`, `BackendState`, and `Self.UserID`.

**Comment by QuantumLibet** [comment]
> /Applications/Tailscale.app/Contents/MacOS/tailscale status --json

Ah! That won't work, since the executable from the AppStore sports a capital 'T' on case-sensitive APFS installations. As in `/Applications/Tailscale.app/Contents/MacOS/Tailscale status --json`. Had me fall on my nose a couple of times already when I used the CLI commands.

```
$ ll /Applications/Tailscale.app/Contents/MacOS/
total 12892
-rwxr-xr-x  1 0    31M  7 Jul 00:56 Tailscale
```


If `macOSTailscalePath` in your linked code does not expect the capital-T-tailscale-executable, then that would explain a lot.

**Comment by rosszurowski** [comment]
Ahhh, shoot. I thought the executable was lowercase. That explains it! I'll make a fix.

Thanks again for reporting and troubleshooting with me.

**Comment by QuantumLibet** [comment]
Thank you for being so nice and helpful. 🫡

I saw #39, but I had quietly hoped you'd fix the App Store application bundle, instead of the Docker Extension. Scripts that run perfectly fine on all sorts of Unixy/Linuxy systems fail miserably. I know how annoying the wait can be for a new App Store release, but then, it would streamline things so much...

**Comment by rosszurowski** [comment]
At this point, I don't think we'd rename that binary because of how people rely on it. We may be able to provide an alias within our bundle if that'd help. But for these scripts that run cross-platform on unix-y systems that fail on macOS, don't you already have to provide a custom path to the script anyways?

(As a side note, our official recommendation for folks using the CLI on macOS is to [set up an alias for easy access](https://tailscale.com/kb/1080/cli/?tab=macos#using-the-cli))

**Comment by QuantumLibet** [comment]
> At this point, I don't think we'd rename that binary because of how people rely on it.

Good point. I hate updates that break my scripts, just as much as anybody else.

An alias is a good solution. It's what they are for. Currently I just add a symbolic link from tailscale to Tailscale, which works fine as well.
